poc-07.py

```diff
@@ -162,9 +162,13 @@ def update(self, theirMessages):
         for theirMessage in theirMessages:
             try:
                 self.driver.update(theirMessage)
-                self.theirState.update(theirMessage) # Would be async.
             except:
                 raise # Would handle or warn.
+      
+        #Their state must be populated with the content of our driver.
+	#FIXME:Later on, update their state with metadata of our Driver messages.
+        for message in self.driver.search():
+            self.theirState.update(message)
```

Comment:
This would harm perfs. We are making a new full read of the other side (twice since from each side)!